dbft.go

```diff
@@ -135,18 +135,19 @@ func (d *DBFT) OnTransaction(tx block.Transaction) {
 	// 	zap.Bool("response_sent", d.ResponseSent()),
 	// 	zap.Bool("block_sent", d.BlockSent()))
 	if !d.IsBackup() || d.NotAcceptingPayloadsDueToViewChanging() ||
-		!d.RequestSentOrReceived() || d.ResponseSent() || d.BlockSent() {
+		!d.RequestSentOrReceived() || d.ResponseSent() || d.BlockSent() ||
+		len(d.MissingTransactions) == 0 {
 		return
 	}
 
-	h := tx.Hash()
-	if _, ok := d.Transactions[h]; ok {
-		return
-	}
-
-	for i := range d.TransactionHashes {
-		if h == d.TransactionHashes[i] {
+	for i := range d.MissingTransactions {
+		if tx.Hash() == d.MissingTransactions[i] {
 			d.addTransaction(tx)
+			if i < len(d.MissingTransactions)-1 {
+				d.MissingTransactions = append(d.MissingTransactions[:i], d.MissingTransactions[i+1:]...)
```

Comment:
If we don't keep them sorted, maybe just swap with the last element?

Reply:
Good idea. In general I don't expect there to be a lot of these, also when we're to find a transaction we'll always remove it, so keeping order seems to be a waste of time.